src/routes/games.js

```diff
@@ -1,56 +1,83 @@
+const { getAllGamesFromDB, addUserGame, removeUserGame, getOnePublicGameByGameID, getAllGamesByUserID, getOnePublicGameByPattern, getAllGameIDsByCategorySearchingPattern, getOneGameByUserID, getAllGamesByEventID, getAllGamesForPlayerInEvent } = require('../db/selectors/games');
 
 module.exports = db => {
   // get all the PUBLIC games from database
-  // this si refered as all game library
-  router.get("/library/games", (req, res) => {
-    // const userId = getLoggedUserId(req);
+  // this is refered as all game library
+  router.get("games/library", (req, res) => {
     getAllGamesFromDB(db)
       .then(data => {
         res.json({ allLibraryGames: data });
       })
   });
 
+  router.post("/user/newGame/:userID/:gameID", (req, res) => {
+    const userID = req.params.userID;
+    const gameID = req.params.gameID;
+    addUserGame(db, gameID, userID)
+      .then(data => {
+        res.json( { message: "successfully added a new game"} );
+    })
+  });
+
+  // remove one game from user's game list
+  router.get("/user/:userID/removeGame/:gameID", (req, res) => {
+    const gameID = req.params.gameID;
+    const userID = req.params.userID;
+    removeUserGame(db, gameID, userID)
+        .then(data => {
+          res.json( { selectedGame: data } );
+        })
+  });
+
   // get one game from public library
-  router.get("/games/:gameID", (req, res) => {
+  router.get("/games/library/:gameID", (req, res) => {
     const gameID = req.params.gameID;
     getOnePublicGameByGameID(db, gameID)
       .then(data => {
-        res.json( {game: data} );
+        res.json( { selectedGame: data } );
       })
   });
 
+    // get all the games owned by a user by given user ID
+    router.get("/user/games/:userID", (req, res) => {
+      const userId = req.params.userID;
+      getAllGamesByUserID(db, userId)
+        .then(data => {
+          res.json( {gamesByUserID: data} );
+        })
+    });
+
   // find games matching the entered pattern in game library
-  router.get("/games/searchNames/:gamePattern", (req, res) => {
+  router.get("/games/library/searchNames/:gamePattern", (req, res) => {
     const gamePattern = req.params.gamePattern;
     console.log(gamePattern);
     getOnePublicGameByPattern(db, gamePattern)
       .then(data => {
-        res.json({ gamesMatchingPattern: data });
+        res.json({ gamesMatchingNamePattern: data });
       })
   });
 
   // find games matching the entered Category pattern in game library
-  router.get("/games/searchCategories/:categorySearchingPattern", (req, res) => {
+  router.get("/games/library/searchCategories/:categorySearchingPattern", (req, res) => {
     const categorySearchingPattern = req.params.categorySearchingPattern;
     getAllGameIDsByCategorySearchingPattern(db, categorySearchingPattern)
       .then(data => {
-        res.json({ games: data });
+        res.json({ gamesMatchingCategoryPattern: data });
```

Comment:
It is not necessary to explain on the keys of the objects you return how you got this data... the developer using this API already knows because of the URL. You should just always name it after what type of data it is... games or game. Otherwise, you will have to be extra careful in the front-end because the API responds with different Keys every time.